File: wlr/OGUR/OGUR/Strategies/ConfusedStrategy.cs
using System;
using OGUR.Creatures;

namespace OGUR.Strategies
{
    public class ConfusedStrategy : IStrategy
    {
        public ConfusedStrategy(ICreature parent)
            : base(parent)
        {
        }
        public override void Act()
        {
            _parent.MoveIfPossible(rand.Next(3) - 1,rand.Next(3) - 1);
        }
    }

    public class ConfusedAndDyingStrategy: ConfusedStrategy
    {
        public ConfusedAndDyingStrategy(ICreature parent)
            : base(parent)
        {
        }
        public override void Act()
        {
            base.Act();
            _parent.ApplyDamage(1.1f,null,false);
        }
    }
}

File: proj/src/aigilas/strategies/impl/ConfusedStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.BaseCreature;
import aigilas.strategies.BaseStrategy;
import aigilas.strategies.Strategy;
import sps.core.RNG;

public class ConfusedStrategy extends BaseStrategy {
    public ConfusedStrategy(BaseCreature parent)

    {
        super(parent, Strategy.Confused);

    }

    @Override
    public void act() {
        _parent.moveIfPossible(RNG.Rand.nextInt(3) - 1, RNG.Rand.nextInt(3) - 1);
    }
}
